### PR TITLE
Solution: push_zero

### PR DESCRIPTION
**Pull request by Utsav1999**
This PR contains solution of the problem <b>push_zero</b>.
Solution includes:

- [X] Solution Approach

- [X] Time and Space Complexity

- [X] Detailed comments of each operation

- [X] Code

- [X] Successfully compiled and tested with given test cases and other test cases.

Thank you. :v:

### CODE REVIEW

File: 1. Array/push_zero/push_zero.cpp [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+* 5
+* 1 0 2 0 3 
+* After performing first for loop : arr[] = {1, 2, 3}, idx = 3 
+* Then no. of zeros present = 5 - 3 = 2 
+* After performing second for loop: arr[] = {1, 2, 3, 0, 0}, idx = 5
+* 
+* Time Complexity:
+* Number of iteration of for loop for non-zero element separation: n
+* Number of iteration of for loop for inserting zeros: (n - idx) = k (let)
+* Final Time Complexity: O(n + k)
```
Comment (Vishal-raj-1) [inline 1. Array/push_zero/push_zero.cpp]: Time Complexity will be O(N), as k is always less than N.

File: 1. Array/push_zero/push_zero.cpp [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+		cout<<a[i]<<" ";
+	}
+	cout<<endl;
+}
+
+// main function
+int main()
+{
+	int t;
+
```
Comment (Vishal-raj-1) [inline 1. Array/push_zero/push_zero.cpp]: unnecessary extra lines.

**Review by Vishal-raj-1 — APPROVED** [review]
@Utsav1999 Your approach is very clear and intuitive. Explanation of code with an example is nice idea. Code is well formatted, try to remove extra lines where it is not necessary. 

Overall, Your work is really nice. Keep Coding 🔥

**Comment by Utsav1999** [comment]
@Vishal-raj-1 thank you for reviewing my code. I will update the solution. 💯

**Comment by Utsav1999** [comment]
@anjali1903 @Vishal-raj-1 please review my work and let me know if any further changes required.
Thank you. ✌️

**Review by Vishal-raj-1 — APPROVED** [review]

**Comment by Vishal-raj-1** [comment]
Great @Utsav1999 . Its Done.

**Comment by Rajwrita** [comment]
> Great @Utsav1999 . Its D

**Comment by Rajwrita** [comment]
> Great @Utsav1999 . Its Done.

Few things: Kindly leave the job of reviewing to the mentors of this repo. 
@Utsav1999 : please accept responses from the mentors of this project only. All else: no.
Thanks

**Comment by Vishal-raj-1** [comment]
@Rajwrita So we can't review peers code ? If there is something incorrect, can't we help each other so that they will not repeat that and learn from that mistake. I was doing this for helping motive.

If you don't want so I will not review anybody else code. Sorry If I break any rule.

Thanks and Regards
Vishal

**Comment by Utsav1999** [comment]
@Rajwrita so do I need to open another PR to get this solution merged?

**Comment by Rajwrita** [comment]
> @Rajwrita So we can't review peers code ? If there is something incorrect, can't we help each other so that they will not repeat that and learn from that mistake. I was doing this for helping motive.
> 
> If you don't want so I will not review anybody else code. Sorry If I break any rule.
> 
> Thanks and Regards
> Vishal

This repo is kept for absolute beginners, and while we totally appreciate your efforts, this feature also makes it difficult for beginners to understand the difference between a correct review and an incorrect one. 

Further details about this repo and all other projects are in winterofcode.com

Feel free to apply as a mentor or a student! Would be amazing :)
Thanks!

**Pull request closed without merging**